### PR TITLE
Migrate CLI Table Utility from oclif v3 to v4 using tty-table

### PR DESCRIPTION
**Pull request by shafeeqd959**
#### **Description:**  
This PR migrates the CLI table implementation from the deprecated `oclif.table` (v3) to `tty-table`, ensuring compatibility with `oclif v4`. The changes introduce a centralized `CLITable` utility, making table rendering more modular and reusable across CLI commands.  

---

#### **Key Changes:**  

✅ **Replaced oclif.table with tty-table**  
- Since `oclif v4` removed support for `oclif.table`, all instances have been updated to use `tty-table`.  

✅ **Implemented `CLITable` as a utility**  
- Added a `CLITable` helper under CLI utilities to standardize table rendering.  
- This utility is now accessible via `cliux.table()`, ensuring a consistent approach across commands.  

✅ **Updated CLI plugins using tables**  
- Refactored the following plugins to use the new `CLITable` utility:  
  - `auth`  
  - `branches`  
  - `audit`  
  - `config`  
  - (and other commands that previously used `oclif.table`)  

✅ **Enhanced table formatting**  
- Introduced **dynamic column formatting** based on field values.  
- Applied color formatting using `chalk` for better readability.  
- Ensured proper text alignment and truncation handling.  

✅ **Refactored existing table logic**  
- Improved handling of **optional headers** (e.g., missing `title`).  
- Used `Map()` instead of `Object.fromEntries()` where applicable for better performance.  
- Ensured multiple value fields are displayed correctly.  

✅ **Maintained backward compatibility**  
- The CLI commands work seamlessly with existing configurations.  
- The `cliux.table()` function maintains the expected behavior while leveraging `tty-table`.  

---

#### **How to Test:**  
1. Run CLI commands that render tables (e.g., `auth list`, `audit logs`, `branches info`).  
2. Verify that tables display correctly with appropriate formatting.  
3. Test with different output flags (`--json`, `--yaml`, `--csv`) to ensure all formats work as expected.  
4. Check edge cases like missing or long text values to confirm truncation and alignment.  

---

#### **Why This Change?**  
- `oclif v4` deprecated `oclif.table`, making it necessary to adopt a new table rendering approach.  
- `tty-table` provides a more flexible and feature-rich table display with better customization.  
- Modularizing table rendering via `CLITable` reduces code duplication and simplifies maintenance.  

---

#### **Impact & Risks:**  
- 🚀 **Improved maintainability & modularity** with a dedicated table utility.  
- ⚠️ **Minor risk of format changes** due to migration—needs validation across CLI commands.  
- ✅ **Thoroughly tested for consistency** to ensure no functional regressions.  

---

#### **Next Steps:**  
- Merge PR after validation across all affected CLI plugins.  
- Update CLI documentation to reflect the new table utility.

### CODE REVIEW

File: packages/contentstack-audit/src/util/flags.ts [hunk 1]
```diff
@@ -1,35 +1,10 @@
...
-//   columns.forEach((element: keyof IFlags) => {
-//     flags[element].helpGroup = 'TABLE';
-
-//     const descriptionKey = `TABLE_${element.toUpperCase()}` as keyof typeof tableColumnDescriptions;
-//     flags[element].description = tableColumnDescriptions[descriptionKey] ?? flags[element].description;
-//   });
-
-//   return flags;
-// }
-
```
Comment (aman19K) [inline packages/contentstack-audit/src/util/flags.ts:32]: @shafeeqd959 , can we remove this commented code if it's not required?

**Review by aman19K — APPROVED** [review]

**Comment by aman19K** [comment]
@shafeeqd959 , LGTM but please remove commented code if it's not required.

File: packages/contentstack-audit/src/util/flags.ts (hunk 1 above)
Comment (shafeeqd959) [reply packages/contentstack-audit/src/util/flags.ts:32]: yes will remove this in the branch

File: packages/contentstack-auth/src/commands/auth/tokens/index.ts [hunk 2]
```diff
@@ -1,19 +1,27 @@
 import { Command } from '@contentstack/cli-command';
-import { cliux, configHandler, formatError } from '@contentstack/cli-utilities';
+import { cliux, configHandler, formatError, CLITable, TableFlags, FlagInput } from '@contentstack/cli-utilities';
 import { BaseCommand } from '../../../base-command';
 export default class TokensListCommand extends BaseCommand<typeof TokensListCommand> {
   static aliases = ['tokens'];
   static examples = ['$ csdx auth:tokens'];
   static description = 'Lists all existing tokens added to the session';
-  // static flags: Record<string, any> = cliux.uxTable.flags(); // use the cli table flags as it displays tokens in table
```
[the remark this answers is not in the document]
Comment (shafeeqd959) [reply packages/contentstack-auth/src/commands/auth/tokens/index.ts:8]: yes this will be removed in the base branch
